src/System.Private.ServiceModel/tests/Common/Infrastructure/Framework.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Infrastructure.Common
+{
+    // Helper class for FrameworkID, an enum created to match
+    // RuntimeInformation.FrameworkDescription
+    public static class Framework
+    {
+        // These literals must match those returned from RuntimeInformation.FrameworkDescription.
+        private const string NetNativeFrameworkName = ".NET Native";
+        private const string NetFrameworkFrameworkName = ".NET Framework";
+        private const string NetCoreFrameworkName = ".NET Core";
+
+        private static FrameworkID _currentFrameworkID = 0;
+        private static string _currentFrameworkDescription;
+
+        private static string CurrentFrameworkDescription
+        {
+            get
+            {
+                if (_currentFrameworkDescription == null)
+                {
+                    _currentFrameworkDescription = RuntimeInformation.FrameworkDescription;
+                }
+
+                return _currentFrameworkDescription;
+            }
+        }
+
+        public static FrameworkID Current
+        {
+            get
+            {
+                if (_currentFrameworkID == 0)
+                {
+                    _currentFrameworkID = DetectCurrentFramework();
+                    Console.WriteLine(String.Format("Detected current framework as \"{0}\" from description \"{1}\"",
+                                     _currentFrameworkID.Name(), CurrentFrameworkDescription));
```

Comment:
Perhaps in a later checkin after we see some VSO runs.  Right now, look at the Console logs on the CI runs and you can see which OS and Framework have been detected.  This would be a great debugging aid if you find tests being run unexpectedly on certain OSes or Frameworks.